lib/filter_lib/src/pagination.py

```diff
@@ -28,18 +28,23 @@ def make_pagination(
     page_num: Optional[int] = None,
 ) -> Tuple[Query, PaginationParams]:
     output_query = initial_query.offset(page_offset).limit(page_size)
-    total_results = initial_query.offset(page_offset).limit(max_count).count()
-    has_more = total_results == max_count
 
-    if total_results == max_count:
-        total_results -= 1
+    total_results_query = initial_query.limit(max_count)
+    if not page_num:  # "page_offset" is used instead of "page_num"
+        total_results_query = total_results_query.offset(page_offset)
+    total_results_count = total_results_query.count()
+
+    has_more = total_results_count == max_count
```

Comment:
Is the situation where total_results_count == max_count == 0 proceeded properly?